### PR TITLE
feat: PubSub for low latency, #29

### LINKED ISSUE #29: PubSub for low latency
Send events from journal over PubSub to the queries. Same as in r2dbc.

### PR DESCRIPTION
**Pull request by patriknw**
Very close 1:1 copy of r2dbc. 

Fixes #29

### CODE REVIEW

File: core/src/main/scala/akka/persistence/dynamodb/internal/PubSub.scala [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+  // Java API
+  def get(system: ActorSystem[_]): PubSub = apply(system)
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class PubSub(system: ActorSystem[_]) extends Extension {
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/dynamodb/internal/PubSub.scala:45]: Doesn't feel great to copy so much code that is the same. We could move this PubSub extension and the EWMA to akka-persistence-typed. Would still be much code, and especially tests that would be duplicates. Don't think there is much we can do about that.

File: core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -7,6 +7,40 @@ akka.persistence.dynamodb {
     # name of the table to use for events
     table = "event_journal"
 
+    # Set this to off to disable publishing of of events as Akka messages to running
```
Comment (pvlugter) [inline core/src/main/resources/reference.conf]:
```suggestion
    # Set this to off to disable publishing of events as Akka messages to running
```

File: core/src/main/resources/reference.conf [hunk 3]
```diff
@@ -7,6 +7,40 @@ akka.persistence.dynamodb {
...
+      # threshold publishing of events is disabled. It is enabled again when lower than
+      # the threshold.
+      throughput-threshold = 400
+      # The interval of the throughput measurements.
+      throughput-collect-interval = 10 seconds
+    }
+
+    # Group the slices for an entity type into this number of topics. Most efficient is to use
+    # the same number as number of projection instances. If configured to less than the number of
+    # of projection instances the overhead is that events will be sent more than once and discarded
```
Comment (pvlugter) [inline core/src/main/resources/reference.conf]:
```suggestion
    # the same number as number of projection instances. If configured to less than the number of
    # projection instances the overhead is that events will be sent more than once and discarded
```

File: core/src/main/scala/akka/persistence/dynamodb/internal/PubSub.scala (hunk 1 above)
Comment (pvlugter) [reply core/src/main/scala/akka/persistence/dynamodb/internal/PubSub.scala:45]: Sounds good to have more of this code shared.

**Review by pvlugter — APPROVED** [review]
LGTM. Reviewed as diffs with the originals.

**Pull request merged**